### DISCUSSION TITLE
[✨] Nested named layout

### DISCUSSION THREAD

**Discussion by motss** [QwikDev/qwik-evolution#164] [Proposals For Qwik]
### Is your feature request related to a problem?

Qwik supports named layout with the following folder structure:

```sh
- src/
 |- routes/
 |   |- layout.tsx # Default parent layout
 |   |- layout-narrow.tsx # Default parent named layout
 |   |- index.tsx # Layout: layout.tsx
 |   `- contact/
 |       `- index@narrow.tsx # Child named layout (Layout: /routes/layout-narrow.tsx)
```

However, it seems like it does not support nested named layout with the following folder structure:

```sh
- src/
 |- routes/
 |   |- layout.tsx # Default parent layout
 |   |- layout-narrow.tsx # Default parent named layout
 |   |- layout-narrow2.tsx # Default parent named layout 2
 |   |- index.tsx # Layout: /routes/layout.tsx
 |   `- contact/
 |        |- index@narrow.tsx # Layout: /routes/layout-narrow.tsx
 |        |- layout.tsx # Default child layout (Layout: /routes/layout.tsx)
 |        |- layout-narrow2@narrow2.tsx # This is not supported (Expected layout: /routes/layout-narrow2.tsx)
 |        |- layout-narrow2.tsx # Default child named layout (Layout: /routes/layout.tsx)
 |        `- contact-1/
 |             `- index@narrow2.tsx # Layout: /routes/contact/layout-narrow2.tsx
```

In the above example, there are two layouts that apply themselves around the `/contact-1` page component:
1. `src/routes/layout-narrow2.tsx`
1. `src/routes/layout-narrow2@narrow2.tsx`

We should be able to define the parent layout of a child layout in Qwik City like `layout-narrow2@narrow2.tsx`. However, as of `qwik-city@0.5.3`, `layout-narrow2@narrow2.tsx` does not work and `/routes/contact/layout-narrow2.tsx` is forced to always render with `/routes/layout.tsx`.

### Describe the solution you'd like

To be able to define a named layout for another named layout as Qwik currently supports only defining a named layout to `index.{tsx,mdx}` instead of a `layout.tsx`, for instance,

## Expected behavior

We should be able to define named layout for layouts.

```sh
# Example 1: Expected rendering flow:
/routes/layout-narrow.tsx
/routes/contact/layout-narrow@narrow.tsx
/routes/contact/contact-1/layout-narrow@narrow.tsx
/routes/contact/contact-1/index@narrow.tsx

# Example 2: Expected rendering flow:
/routes/layout-narrow.tsx
/routes/contact/layout@narrow.tsx
/routes/contact/contact-1/layout-narrow.tsx
/routes/contact/contact-1/index@narrow.tsx

# Example 3: Expected rendering flow:
/routes/layout.tsx
/routes/contact/layout.tsx
/routes/contact/contact-1/layout-narrow.tsx
/routes/contact/contact-1/index@narrow.tsx

# Example 4: Expected rendering flow:
/routes/layout.tsx
/routes/contact/layout-narrow.tsx
/routes/contact/contact-1/layout@narrow.tsx
/routes/contact/contact-1/index.tsx
```

## Actual behavior

We can only specify named layout in an `index.{tsx,mdx}` file like `index@narrow.tsx` and `layout-narrow.tsx` is always defaulting to a `layout.tsx` at upper level.

```sh
# Actual rendering flow:
/routes/layout.tsx
/routes/contact/layout.tsx
/routes/contact/contact-1/layout-narrow.tsx
/routes/contact/contact-1/index@narrow.tsx
```

### Describe alternatives you've considered

Avoid using nested named layout

**Comment by wmertens** [comment]
@motss I don't understand, what is the use case? How should it detect what name to use?

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
